### DISCUSSION TITLE
Spam handling

### DISCUSSION THREAD

**Discussion by tassoman** [misskey-dev/misskey#9190] [General]
Any idea how to prevent spamming?
There are activitypub spammers who implemented a way to spam instances each-others.
When I try to deactivate or delete any of this accounts (b8b, 33bet, s666bet, dubaipalace), Delivery Queue gets flooded with 250 events. Then I found there are 10-13 delayed activities for other tens of instances (for about 124 delayed outgoing activity)

I guess they have developed something who spams other instances on deletion event. So instances are spamming each other.

I found #7091 #7088 #7089 #7090 issues related to spam, can be related

![immagine](https://user-images.githubusercontent.com/92727/202433308-1b1c8486-3cf2-4cc3-8d03-02c6a8f99b62.png)

![immagine](https://user-images.githubusercontent.com/92727/202433897-16622fe1-620d-43f1-895b-5652309092c6.png)

**Comment by ThatOneCalculator** [comment]
Jobs can also get stuck if the instances no longer exist and thus cannot complete. For example, `mk.disqordia.space` is no longer in service. The Misskey forks Calckey and Foundkey clear job queues from dead instances unlike Misskey.

**Reply by tassoman** [reply]
I realized jobs went to that instances while Instances were un reachable. I can't say if I flooded them or simply they were offline for own problems :thinking: 
BTW the activity was contextual of profile deletion. Profiles weren't following and weren't followed by anyone. I can't say the reason why of such that activity

**Reply by tassoman** [reply]
maybe this was an effect of forkbomb vunerability just discovered some days ago? :thinking:

**Reply by ThatOneCalculator** [reply]
Doubt it.

**Comment by tassoman** [comment]
Sorry for digging an old post.  
How do I can delete nowadays spammy users, messages and attachments?
Spammy account can only be suspended (2024.02.0)

**Reply by tamaina** [reply]
Deleting the user should erase everything.

**Reply by tassoman** [reply]
Yes, indeed. I confirm user deletion can delete everything from that user.  
The feature is only available to administrators, no user deletion for moderators.  
It's a first step for manually spam deletion.
More on this, today an [antispam filter is under deployment](https://github.com/misskey-dev/misskey/pull/13343).

**Comment by tassoman** [comment]
Anti mention spam handling can be configured using roles
